pennylane/ops/functions/dot.py

```diff
@@ -90,6 +104,32 @@ def dot(
         <lambda>(params_0, t) * X(0)
       + <lambda>(params_1, t) * Y(0)
     )
+
+    .. details::
+        :title: Grouping
+
+        Grouping information can be collected during construction using the ``grouping_type`` and ``method``
+        keyword arguments. For example:
+
+        .. code-block:: python
+
+            import pennylane as qml
+
+            a = qml.X(0)
+            b = qml.prod(qml.X(0), qml.X(1))
+            c = qml.Z(0)
+            obs = [a, b, c]
+            coeffs = [1.0, 2.0, 3.0]
+
+            op = qml.dot(coeffs, obs, grouping_type="qwc")
+
+        >>> op.grouping_indices
+        ((2,), (0, 1))
+
+        ``grouping_type`` can be ``"qwc"`` (qubit-wise commuting), ``"commuting"``, or ``"anticommuting"``, and
+        ``method`` can be ``"rlf"`` or ``"lf"``. To see more details about how these affect grouping, check out the
+        `Pauli grouping <https://docs.pennylane.ai/en/stable/code/qml_pauli.html#module-pennylane.pauli.grouping.graph_colouring>`_
```

Comment:
Can we get this to be a dynamic rather than static link?

Reply:
I'm not sure how to do that as we're referring to a section of a `.rst` file rather than a function or class. Could you add it as a suggestion if you know how to do it?